File: web/src/test/java/no/nav/modiapersonoversikt/legacy/varsel/service/VarslerServiceImplTest.java
package no.nav.modiapersonoversikt.legacy.varsel.service;

import no.nav.modiapersonoversikt.legacy.varsel.domain.Varsel;
import no.nav.tjeneste.virksomhet.brukervarsel.v1.BrukervarselV1;
import no.nav.tjeneste.virksomhet.brukervarsel.v1.HentVarselForBrukerUgyldigInput;
import no.nav.tjeneste.virksomhet.brukervarsel.v1.meldinger.WSHentVarselForBrukerRequest;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import javax.xml.ws.soap.SOAPFaultException;
import java.util.List;
import java.util.Optional;

import static org.junit.Assert.assertTrue;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class VarslerServiceImplTest {

    @Mock
    private BrukervarselV1 brukervarsel;

    @InjectMocks
    VarslerServiceImpl impl;

    @Test
    public void skalIkkeTryneHeleVerdenOmDetSkjerSoapFaults() throws HentVarselForBrukerUgyldigInput {
        when(brukervarsel.hentVarselForBruker(any(WSHentVarselForBrukerRequest.class))).thenThrow(SOAPFaultException.class);
        Optional<List<Varsel>> varsler = impl.hentAlleVarsler("10108000398");

        assertTrue(varsler.isEmpty());
    }
}

File: web/src/test/java/no/nav/modiapersonoversikt/legacy/varsel/service/VarslerServiceImplTest.kt
package no.nav.modiapersonoversikt.legacy.varsel.service

import io.mockk.every
import io.mockk.mockk
import no.nav.common.types.identer.Fnr
import no.nav.modiapersonoversikt.consumer.brukernotifikasjon.Brukernotifikasjon
import no.nav.modiapersonoversikt.service.varsel.VarslerService
import no.nav.modiapersonoversikt.service.varsel.VarslerServiceImpl
import no.nav.tjeneste.virksomhet.brukervarsel.v1.BrukervarselV1
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import javax.xml.soap.SOAPFault
import javax.xml.ws.soap.SOAPFaultException

class VarslerServiceImplTest {
    val brukervarselV1 = mockk<BrukervarselV1>()
    val brukernotifikasjonService = mockk<Brukernotifikasjon.Service>()
    val soapfault = mockk<SOAPFault>()

    val varselService: VarslerService = VarslerServiceImpl(brukervarselV1, brukernotifikasjonService)

    @Test
    internal fun `skal ikke tryne hele verden om det skjer soap faults`() {
        every { soapfault.faultString } returns ""
        every { brukervarselV1.hentVarselForBruker(any()) } throws SOAPFaultException(soapfault)
        val varsler = varselService.hentAlleVarsler(Fnr("12345678910"))
        assertThat(varsler).isEmpty()
    }
}
